src/bosh_openstack_cpi/lib/cloud/openstack/cloud.rb

```diff
@@ -777,7 +777,9 @@ def validate_options
         }
         if Bosh::OpenStackCloud::Openstack.is_v3(auth_url)
           openstack_options_schema['openstack']['project'] = String
-          openstack_options_schema['openstack']['domain'] = String
+          openstack_options_schema['openstack'][optional('domain')] = String
+          openstack_options_schema['openstack'][optional('user_domain_name')] = String
+          openstack_options_schema['openstack'][optional('project_domain_name')] = String
```

Comment:
Maybe try something like this instead:
```ruby
          if @options['openstack']['user_domain_name'] || @options['openstack']['project_domain_name']
            openstack_options_schema['openstack']['user_domain_name'] = String
            openstack_options_schema['openstack']['project_domain_name'] = String
          else
            openstack_options_schema['openstack']['domain'] = String
          end
```
It should be safe to check for domain, since the existence of the openstack key has been checked [here](https://github.com/cloudfoundry/bosh-openstack-cpi-release/blob/8115ba138b82b82035f1337da1ae28ed6dc8f7d2/src/bosh_openstack_cpi/lib/cloud/openstack/cloud.rb#L747-L748). The above example we would not have to change the current behaviour when someone forgets to specify a domain when using the v3 api.